onnxruntime/core/providers/cpu/activation/element_wise_ranged_transform.h

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include "core/common/common.h"
+#include "core/platform/threadpool.h"
+#include "core/framework/op_kernel.h"
+#include "core/util/math_cpuonly.h"
+
+namespace onnxruntime {
+namespace functors {
+
+inline common::Status GetFloatParam(const std::string& name, const onnxruntime::NodeAttributes& attributes,
+                                    float& out) {
+  auto attr = attributes.find(name);
+  if (attr == attributes.end()) {
+    return ORT_MAKE_STATUS(ONNXRUNTIME, FAIL, "No attribute with name:'", name, "'is defined.");
+  }
+  if (attr->second.type() != ONNX_NAMESPACE::AttributeProto::AttributeType::AttributeProto_AttributeType_FLOAT) {
+    return Status(ONNXRUNTIME, FAIL, "Attibute name and type don't match");
+  }
+  out = attr->second.f();
+  return common::Status::OK();
+}
+
+// Like the std::transform
+// T should be float or double
+template <typename T>
+struct ElementWiseRangedTransform {
+  using DataType = T;
+  const T* input = nullptr;
+  T* output = nullptr;
+  // Run an unary function through the range [input + first, input + last) -> [output + first, output + last)
+  // Thread safe
+  virtual void operator()(std::ptrdiff_t first, std::ptrdiff_t last) const = 0;
+  // Thread safe
+  virtual float Cost() const = 0;
+  // Return a deep clone of *this
+  virtual ElementWiseRangedTransform<T>* Copy() const = 0;
+  virtual ~ElementWiseRangedTransform() = 0;
+
+  // A helper function for creating such objects by type name.
+  static Status Create(const std::string& type, const onnxruntime::NodeAttributes& attributes,
```

Comment:
I found the param 'type' a bit confusing at first. May be call it transform_type/transform_name/activation_type? 
Also, would be good to return a unique_ptr as out param to avoid memory leaks elsewhere.